clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -473,12 +473,30 @@
                                                                        new Field("max_bytes",
                                                                                  INT32,
                                                                                  "Maximum bytes to fetch."));
+    // FETCH_REQUEST_PARTITION_V1 added log_start_offset field - the earliest available offset of partition data that can be consumed.
```

Comment:
Good point. Maybe we should specify this in the explanation of the field instead of in the comment? I added following to the field's doc: The field is only used when request is sent by follower. For simplicity I didn't specify that the field will be set to -1 if the request is sent by consumer.